### PR TITLE
hash/ibm5150.xml: first pass at QA

### PR DESCRIPTION
**Pull request by angelosa**
- Revisit all of the supported="partial" / "no" flags, promote or demote depending on observations.

Software list items promoted to working
---------------------------------------
ibm5150.xml: Ancient Land of Ys [Angelo Salese]

### CODE REVIEW

**Comment by Allofich** [comment]
There's something wrong with "Big Top". I suspect it may be a bad dump. The game prompts you to press a number from 1 to 9 to choose difficulty, with 9 being the hardest. If you press "1", the game plays at an unplayable crawl. Any other number produces a more normal speed.

When you do this, the game pulls a 2-byte value from an array, which affects the game speed. The higher the number, the more the game is delayed. The number pressed is used as an index into the array.
```
1     2     3     4     5     6     7     8     9         Number pressed (difficulty)
0B 80 07 00 05 00 00 03 00 02 80 01 00 01 80 00 01 00     Array of values
```
"1" will retrieve 0B80, or 32779, by far the largest value, making the game crawl. "4" will also retrieve a somewhat large value of 0300, or 768, and so actually plays more slowly than difficulties 2 or 3.

There seems to be two problems. You can find what seems to be this array data at offset 622A in `big top.img` as follows:

```
1     2     3     4     5     6     7     8     9         1-based index
0B 80 07 00 05 00 03 00 02 80 01 00 01 80 00 01 00 00     Array of values
```
which looks more sensible, especially if the "80" values were actually "00" values.

```
1     2     3     4     5     6     7     8     9         1-based index
0B 00 07 00 05 00 03 00 02 00 01 00 01 00 00 01 00 00     Array of values
```
In-game, though, there is an extra "00" in there, which makes the values for difficulties 4 and up strange. At offset 133F9 in the file, there is what seems to be the same array but with the extra "00" attached.

```
0B 80 07 00 05 00 00
```
after which is just a bunch of "F6" values. It seems the game attaches the latter part of the earlier data to the offset 133F9 data to create the array. (Testing supports this. I can affect the speeds of the in-game difficulties for difficulties 1-3 by hex-editing the 133F9 data (but no effect by editing the 622A data), and the speeds for difficulties 4 and up by adjusting the 622A data)

There might just be a programming error in the original game, but it does at least seem like the "80" values may be errors in the dump, since changing them to 00 will fix the speed of difficulty 1, which is clearly wrong since it's unplayable.

**Comment by Allofich** [comment]
Tried a different dump of "Big Top", and it has neither of the above problems (difficulty 1 runs fine, difficulty 4 is faster than the lower difficulties).

File: hash/ibm5150.xml [hunk 1]
```diff
@@ -15184,7 +15192,15 @@ Fatal error: Incorrect layout on track 39 head 0, expected_size=100000, current_
 		<description>The Software Toolworks Game Pack 3</description>
 		<year>1993</year>
 		<publisher>The Software Toolworks</publisher>
+		<notes><![CDATA[
+Jack Nicklaus Ultimate Golf & Course Design:
+Road &amp; Track Presents Grand Prix Unlimited: goes autopilot when using keyboard (verify)
+F-19 Stealth Fighter: sports copy protection but it's skipped?
+Life &amp; Death -- You are the Surgeon:
+Gin King and Cribbage King: prints "packet file is corrupt after selection in batch file, both games shows a bit of [VGA] offset [pel_shift]
```
Comment (Allofich) [inline hash/ibm5150.xml:15200]: I think you meant to put a closing quote after "corrupt" here

**Comment by angelosa** [comment]
> Tried a different dump of "Big Top", and it has neither of the above problems (difficulty 1 runs fine, difficulty 4 is faster than the lower difficulties).

May be earlier versioning as well, I'll update the notes field.

**Pull request merged**